src/RRule.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+namespace ipl\Scheduler;
+
+use BadMethodCallException;
+use DateTime;
+use DateTimeInterface;
+use Generator;
+use ipl\Scheduler\Contract\Frequency;
+use Recurr\Rule as RecurrRule;
+use Recurr\Transformer\ArrayTransformer;
+use Recurr\Transformer\ArrayTransformerConfig;
+use Recurr\Transformer\Constraint\AfterConstraint;
+
+use function ipl\Stdlib\get_php_type;
+
+class RRule implements Frequency
+{
+    /** @var string Run once a year */
+    public const YEARLY = 'YEARLY';
+
+    /** @var string Run every 3 month starting from the given start time */
+    public const QUARTERLY = 'QUARTERLY';
+
+    /** @var string Run once a month */
+    public const MONTHLY = 'MONTHLY';
+
+    /** @var string Run once a week based on the specified start time */
+    public const WEEKLY = 'WEEKLY';
+
+    /** @var string Run once a day at the specified start time */
+    public const DAILY = 'DAILY';
+
+    /** @var string Run once an hour */
+    public const HOURLY = 'HOURLY';
+
+    /** @var string Run once a minute */
+    public const MINUTELY = 'MINUTELY';
+
+    /** @var int Maximum time difference between the current date and the start date in years */
+    public const MAX_YEAR_DIFF = -5;
+
+    /** @var RecurrRule */
+    protected $rrule;
+
+    /** @var ArrayTransformer */
+    protected $transformer;
+
+    public function __construct(string $rule, DateTimeInterface $start)
+    {
+        $this->rrule = new RecurrRule($rule);
+        $this->transformer = new ArrayTransformer();
+
+        $this->startAt($start);
+        $this->limitTransformer(3);
```

Comment:
Why 3 and not just 1?

Reply:
Usually we are rendering **3** next recurrence, so I was thinking of having this as a default as well, then the forms don't have to set the limit on their own.